ledger-core/server/server.go

```diff
@@ -22,6 +23,24 @@ func NewNode(cfg configuration.Configuration) Server {
 	return insapp.New(cfg, appFactory)
 }
 
+func NewMultiServerWithoutPulsar(configProvider *CloudConfigurationProvider) (Server, pulsar.SelectivePulseDistributor) { // nolint:interfacer
```

Comment:
Its better not to build another contstructor. You can do something like this:
```
func NewMultiServer(configProvider *CloudConfigurationProvider, extraComponents ...interface{}) Server {
	controller := cloud.NewController()
	if configProvider.GetAppConfigs == nil {
		panic("GetAppConfigs cannot be nil")
	}

	multiFn := func(provider insapp.ConfigurationProvider) ([]configuration.Configuration, insapp.NetworkInitFunc) {
		conf := provider.(*CloudConfigurationProvider)
		return conf.GetAppConfigs(), controller.NetworkInitFunc
	}

	return insapp.NewMulti(
		configProvider,
		appFactory,
		multiFn,
		extraComponents...,
	)
}
```

So pulsar would be optional for running MultiServer

Reply:
Why is it better? It's just one more constructor. I don't think this version is more convenient.

Reply:
this allow to add more components, and lowers copypasted code

Reply:
1. Do we need it?
2. I need to return controller or somehow join it one of the extraComponents ( to have opportunity to change pulses )

Reply:
1. yes, it should not be 10 of copypasted constructors
2. you can create controller outside, it will be started with server and could be used outside

Reply:
1. They are different
2. And at that case I will need to create controller outside every time? Even when I don't need it outside?

Reply:
Controller is in 'internal' dir

Reply:
1. this is poor design. The name is NewMultiServerWithoutPulsar, why does it return controller? 
It could be NewControlledMultiServer() with external controller, that can manage pulses
2. If you start NewMultiServer than yoo don't need controller
3. It could be moved from internal to testutils

Reply:
1. It doesn't return controller. It returns SelectivePulseDistributor
2. yes, but you sed that controller should be created outside and forwarded to constructor

Reply:
1. I don't understand why. Because controller could be used not only as PulseDistributor (I'm working on it)
2. I'm against NewMultiServerWithoutPulsar, but NewControlledMultiServer with controller as argument looks OK

Reply:
ок